aten/src/ATen/native/cpu/ComplexKernel.cpp

```diff
@@ -0,0 +1,32 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/TensorFactories.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+
+namespace at {
+namespace native {
+namespace {
+
+void complex_kernel(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.input_dtype(), "complex_cpu", [&]() {
+    cpu_kernel(iter, [=](scalar_t a, scalar_t b) -> c10::complex<scalar_t> {
+      return c10::complex<scalar_t>(a, b);
+    });
+  });
+}
+
+void complex_polar_kernel(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES(iter.input_dtype(), "complex_polar_cpu", [&]() {
+    cpu_kernel(iter, [=](scalar_t a, scalar_t b) -> c10::complex<scalar_t> {
+      return c10::complex<scalar_t>(a * std::cos(b), a * std::sin(b));
```

Comment:
I am intellectually curious if there is a faster way to compute this since we want both the cosine and sine of `b`, but I don't think it's interesting to try and compute sine as `sqrt(1 - (std::cos(b))^2`

Reply:
There will have to be checks for sign though if sqrt is used right. I changed the kernel to this (has sign errors)
```c++
void complex_polar_kernel(TensorIterator& iter) {
  AT_DISPATCH_FLOATING_TYPES(iter.input_dtype(), "complex_polar_cpu", [&]() {
    cpu_kernel(iter, [=](scalar_t a, scalar_t b) -> c10::complex<scalar_t> {
      scalar_t acos = a * std::cos(b);
      scalar_t asin = std::sqrt(a * a - acos * acos);
      return c10::complex<scalar_t>(acos, asin);
    });
  });
}
```
and I ran this
```python
import time
import torch

a = torch.randn(100, 100)
b = torch.randn(100, 100)

start = time.time()
for i in range(50000):
    z = torch.complex_polar(a, b)
end = time.time()
print(end - start)
```
and the difference was 3.96 (current) vs 3.69 (sqrt(1-cos^2)) seconds on my computer.

Reply:
Thanks for investigating - that's cool! And yes, there will be sign issues because the solution to the problem is ambiguous. Interesting that it would seem to make a significant performance improvement, though, if the sign issue could be resolved. I don't think we need to be so performance sensitive with this PR, however.

Reply:
Interesting I think we can do something like this:
```
void complex_polar_kernel(TensorIterator& iter) {
  AT_DISPATCH_FLOATING_TYPES(iter.input_dtype(), "complex_polar_cpu", [&]() {
    cpu_kernel(iter, [=](scalar_t a, scalar_t b) -> c10::complex<scalar_t> {
      scalar_t cos_val = std::cos(b);
      scalar_t sin_val = std::sqrt(1 - cos_val * cos_val);
      if (b < 0) {
         sin_val = sin_val * -1;
      }
      return c10::complex<scalar_t>(a*cos_val, a*sin_val);
    });
  });
}
```

@wjgan7 can you check if there's a performance boost for this?